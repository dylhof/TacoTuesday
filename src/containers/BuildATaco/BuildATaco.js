import React, {Component} from 'react';
import { connect } from 'react-redux';
import {BuildATacoPart} from '../BuildATacoPart/BuildATacoPart';
import uuid from 'uuid/v4';

export class BuildATaco extends Component{

  tacoChoices = () => {
    const {baseLayers, mixins, seasonings, condiments, shells } = this.props
    const tacoParts = [baseLayers, mixins, seasonings, condiments, shells]
    const tacoPartName = ['Base Layer', 'Mixin', 'Seasoning', 'Condiment', 'Shell']
    const jsxTacoChoices = tacoParts.map((part, i) => <BuildATacoPart key={uuid()} tacoPart={part} tacoPartName={tacoPartName[i]}/>)
    return jsxTacoChoices
  }

  render() {
    return(
      <div>
        {this.tacoChoices()}
      </div>
    )
  }
}

export const mapStateToProps = state => ({
  baseLayers: state.baseLayers,
  mixins: state.mixins,
  seasonings: state.seasonings,
  condiments: state.condiments,
  shells: state.shells
})

export default connect(mapStateToProps)(BuildATaco)