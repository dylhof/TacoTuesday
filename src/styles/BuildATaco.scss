.buildATacoDiv {
  @include divFlex(flex-start, space-around, row);
  background-color: $highlight;
  opacity: .9;
  height: 100vh;
  margin: 5px;
  border-radius: 5px
}