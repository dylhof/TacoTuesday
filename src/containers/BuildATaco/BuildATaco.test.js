import React from 'react';
import { shallow } from 'enzyme';
import {BuildATaco, mapStateToProps} from './BuildATaco';
import uuid from 'uuid/v4';

jest.mock('uuid/v4');
uuid.mockReturnValue(1);

describe('BuildATaco', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(
      <BuildATaco/>
    )
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  describe('mapStateToProps', () => {
    it('should return an object with all correct props', () => {
      //setup
      const mockState = {
        baseLayers: [{name: 'taco!'}],
        mixins: [{name: 'taco!'}],
        condiments: [{name: 'taco!'}],
        seasonings: [{name: 'taco!'}],
        shells: [{name: 'taco!'}],
        error: 'Sorry no tacos'
      }
      const expected = {
        baseLayers: [{name: 'taco!'}],
        mixins: [{name: 'taco!'}],
        condiments: [{name: 'taco!'}],
        seasonings: [{name: 'taco!'}],
        shells: [{name: 'taco!'}],
      }
      //execution
      const mappedProps = mapStateToProps(mockState)
      //expectation
      expect(mappedProps).toEqual(expected)
    })
  })
})