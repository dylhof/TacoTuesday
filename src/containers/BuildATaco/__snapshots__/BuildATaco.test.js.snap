// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuildATaco should match the snapshot 1`] = `
<div>
  <BuildATacoPart
    key="1"
    tacoPartName="Base Layer"
  />
  <BuildATacoPart
    key="1"
    tacoPartName="Mixin"
  />
  <BuildATacoPart
    key="1"
    tacoPartName="Seasoning"
  />
  <BuildATacoPart
    key="1"
    tacoPartName="Condiment"
  />
  <BuildATacoPart
    key="1"
    tacoPartName="Shell"
  />
</div>
`;
